src/MessageReader.js

```diff
@@ -39,21 +39,36 @@ class StandardTypeReader {
     this.view = new DataView(buffer.buffer, buffer.byteOffset);
   }
 
-  string() {
+  string(isJson: boolean) {
     const len = this.int32();
     const codePoints = new Uint8Array(this.buffer.buffer, this.buffer.byteOffset + this.offset, len);
     this.offset += len;
     // if the string is relatively short we can use apply
     // but very long strings can cause a stack overflow due to too many arguments
     // in those cases revert to a slower itterative string building approach
     if (codePoints.length < 1000) {
-      return String.fromCharCode.apply(null, codePoints);
+      const resultString = String.fromCharCode.apply(null, codePoints);
+      if (isJson) {
+        try {
+          return JSON.parse(resultString);
+        } catch {
+          return `Could not parse ${resultString}`;
+        }
+      }
+      return resultString;
     }
 
     let data = "";
     for (let i = 0; i < len; i++) {
       data += String.fromCharCode(codePoints[i]);
     }
+    if (isJson) {
+      try {
+        return JSON.parse(data);
+      } catch {
+        return `Could not parse ${data}`;
+      }
+    }
```

Comment:
Maybe deduplicate this code by having `resultString` fall through instead of an early return?